src/figuresClasses.ts

```diff
@@ -1,11 +1,75 @@
-export interface Figure {}
+type Shape = `triangle` | `circle` | `rectangle`;
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides: number[] = [this.a, this.b, this.c];
+    const maxSide = Math.max(...sides);
+    const sumOfTwoSides = sides.reduce((sum, side) => sum + side, 0) - maxSide;
+
+    if (sides.some((side) => side <= 0) || sumOfTwoSides <= maxSide) {
+      throw new Error('your error message');
```

Comment:
Consider providing a more descriptive error message here, such as 'Invalid triangle sides: the sum of any two sides must be greater than the third side.'